app/utils/gitlab_manager.py

```
@@ -1,42 +1,42 @@
-            response = self.session.get(url, headers=self.headers)
+            response = self._rq.get(url, headers=self._header)
             response.raise_for_status()
             return response.json()
         except requests.exceptions.RequestException as e:
-            print(f"Error fetching repos: {e}")
-            return False
+            raise DevDoxAPIException(
+                user_message="Unable to fetch GitLab user.",
+                log_message="GitLab user fetch failed",
+                root_exception=e,
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)